tests/audio/test_pit.py

```diff
@@ -0,0 +1,210 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+from functools import partial
+from typing import Callable, Tuple
+
+import numpy as np
+import pytest
+import torch
+from scipy.optimize import linear_sum_assignment
+from torch import Tensor
+
+from tests.helpers import seed_all
+from tests.helpers.testers import BATCH_SIZE, NUM_BATCHES, MetricTester
+from torchmetrics.audio import PIT
+from torchmetrics.functional import pit, si_sdr, snr
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_6
+
+seed_all(42)
+
+TIME = 10
+
+Input = namedtuple('Input', ["preds", "target"])
+
+# three speaker examples to test _find_best_perm_by_linear_sum_assignment
+inputs1 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, TIME),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, TIME),
+)
+# two speaker examples to test _find_best_perm_by_exhuastive_method
+inputs2 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, TIME),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, TIME),
+)
+
+
+def a_naive_implementation_of_pit_based_on_scipy(
+    preds: Tensor,
+    target: Tensor,
+    metric_func: Callable,
+    eval_func: str,
+) -> Tuple[Tensor, Tensor]:
+    """a naive implementation of pit based on scipy
```

Comment:
do you have a reference for this or did you just implement it yourself?

Reply:
It's just implemented by myself. The idea of PIT is very simple: find the best (minimal or maximal) metric among all possible permutations of preds and target. Thus, it is equivalent to calculate the metric matrix first, and then solve the linear sum assignment problem.

[pytorch-pit](https://github.com/asteroid-team/pytorch-pit) is the pit used in [asteroid](https://github.com/asteroid-team/asteroid).  Should I change to use it to verify my implementation?

Reply:
No, I am fine with this explanation that you just gave. Maybe you could add it to the docstring of the function?

Reply:
could you pls add the same explanation to the function docstring?